cmd/segment-reaper/observer.go

```diff
@@ -143,11 +145,88 @@ func (obsvr *observer) processSegment(ctx context.Context, path metainfo.ScopedP
 // analyzeProject analyzes the objects in obsv.objects field for detecting bad
 // segments and writing them to objs.writer.
 func (obsvr *observer) analyzeProject(ctx context.Context) error {
-	// TODO this part will be implemented in next PR
-	// TODO(if): For what is this?
+	for bucket, objects := range obsvr.objects {
+		for path, object := range objects {
+			if object.skip {
+				continue
+			}
+
+			brokenObject := false
+			if object.hasLastSegment {
+				if object.expectedNumberOfSegments == 0 {
+					if !object.segments.IsSequence() {
+						brokenObject = true
+					}
+				} else if object.segments.Count() != int(object.expectedNumberOfSegments)-1 {
+					// TODO should we also check if its valid sequence
+					// expectedNumberOfSegments-1 because 'segments' doesn't contain last segment
+					brokenObject = true
+				}
+
+				if brokenObject {
+					err := obsvr.printSegment(ctx, "l", bucket, path, obsvr.writer)
+					if err != nil {
+						return err
+					}
+				}
+			} else {
+				brokenObject = true
+			}
+
+			if brokenObject {
+				for index := 0; index < maxNumOfSegments; index++ {
+					has, err := object.segments.Has(index)
+					if err != nil {
+						return nil
+					}
+					if has {
+						err := obsvr.printSegment(ctx, "s"+strconv.Itoa(index), bucket, path, obsvr.writer)
+						if err != nil {
+							return err
+						}
+					}
+				}
+			}
+		}
+	}
 	return nil
 }
 
+func (obsvr *observer) printSegment(ctx context.Context, bucket string, segmentIndex string, path string, csvWriter *csv.Writer) error {
+	creationDate, err := pointerCreationDate(ctx, obsvr.db, obsvr.lastProjectID, segmentIndex, bucket, path)
+	if err != nil {
+		return err
+	}
+	encodedPath := base64.StdEncoding.EncodeToString([]byte(path))
+	err = csvWriter.Write([]string{
+		obsvr.lastProjectID,
+		segmentIndex,
+		bucket,
+		encodedPath,
+		creationDate,
+	})
+	if err != nil {
+		return err
+	}
+	obsvr.zombieSegments++
+	return nil
+}
+
+func pointerCreationDate(ctx context.Context, db metainfo.PointerDB, projectID string, bucket string, segmentIndex string, path string) (string, error) {
+	key := []byte(storj.JoinPaths(projectID, segmentIndex, bucket, path))
+	pointerBytes, err := db.Get(ctx, key)
+	if err != nil {
+		return "", err
+	}
+
+	pointer := &pb.Pointer{}
+	err = proto.Unmarshal(pointerBytes, pointer)
+	if err != nil {
+		return "", err
+	}
+	return pointer.CreationDate.String(), nil
```

Comment:
Let's ensure that we use the same format for writing the date in the CSV file and for reading it.

I got errors of parsing the dates in the CSV file with the `delete` command under 


For reading, we use [`RFC3339`](https://github.com/storj/storj/pull/3660/files#diff-e8fb83f0bab533d6fafa0225792233b5R84) and here we should use the same with [`Format` method](https://golang.org/pkg/time/#Time.Format).

On the other hand, I think that it would be better to use `RFC3339Nano` format.

Reply:
done